test_integration/test_utils.go

```diff
@@ -0,0 +1,362 @@
+	}
+	defer logs.Close()
+
+	// Print the logs
+	if _, err := io.Copy(os.Stdout, logs); err != nil {
+		return fmt.Errorf("compiling bacalhau: failed to print logs: %w", err)
+	}
+
+	// Copy the compiled binary from the container to the host
+	reader, err := container.CopyFileFromContainer(ctx, "/usr/src/bacalhau-binary")
+	if err != nil {
+		return fmt.Errorf("compiling bacalhau: failed to copy file from container: %w", err)
+	}
+	defer reader.Close()
+
+	// Create the output file on the host
+	outFile, err := os.Create(filepath.Join(programDir, "test_integration", "assets", "dockerfiles", "bacalhau_bin"))
+	if err != nil {
+		return fmt.Errorf("compiling bacalhau: failed to create output file: %w", err)
+	}
+	defer outFile.Close()
+
+	// Copy the content from the reader to the output file
+	_, err = io.Copy(outFile, reader)
+	if err != nil {
+		return fmt.Errorf("compiling bacalhau: failed to write compiled program to file: %w", err)
+	}
+
+	fmt.Println("Compiled program has been copied to the host machine.")
+
+	return nil
+}
+
+func buildBaseImages(testIdentifier string) error {
+	err := buildDockerImage(
+		"./assets/dockerfiles/Dockerfile-JumpboxNode",
+		"bacalhau-test-jumpbox-"+testIdentifier,
+		testIdentifier,
+	)
+	if err != nil {
+		return fmt.Errorf("error creating the bacalhau-test-jumpbox image: %v", err)
+	}
+
+	err = buildDockerImage(
+		"./assets/dockerfiles/Dockerfile-ComputeNode",
+		"bacalhau-test-compute-"+testIdentifier,
+		testIdentifier,
+	)
+	if err != nil {
+		return fmt.Errorf("error creating the bacalhau-test-compute image: %v", err)
+	}
+
+	err = buildDockerImage(
+		"./assets/dockerfiles/Dockerfile-DockerImageRegistryNode",
+		"bacalhau-test-registry-"+testIdentifier,
+		testIdentifier,
+	)
+	if err != nil {
+		return fmt.Errorf("error creating the bacalhau-test-registry image: %v", err)
+	}
+
+	err = buildDockerImage(
+		"./assets/dockerfiles/Dockerfile-RequesterNode",
+		"bacalhau-test-requester-"+testIdentifier,
+		testIdentifier,
+	)
+	if err != nil {
+		return fmt.Errorf("error creating the bacalhau-test-requestor image: %v", err)
+	}
+
+	return nil
+}
+
+func setTestGlobalEnvVariables(additionalSetupEnvVars map[string]string) error {
+	defaultEnvVars := map[string]string{
+		"TESTCONTAINERS_RYUK_DISABLED":                  "false",
+		"TESTCONTAINERS_RYUK_CONTAINER_PRIVILEGED":      "true",
+		"TESTCONTAINERS_RYUK_CONTAINER_STARTUP_TIMEOUT": "7m",
+		"TESTCONTAINERS_RYUK_CONNECTION_TIMEOUT":        "10m",
+		"TESTCONTAINERS_RYUK_RECONNECTION_TIMEOUT":      "7m",
+		"TESTCONTAINERS_RYUK_VERBOSE":                   "true",
+		"DOCKER_API_VERSION":                            "1.45",
+	}
+
+	// Merge additional env vars with default ones
+	if additionalSetupEnvVars != nil {
+		for key, value := range additionalSetupEnvVars {
+			defaultEnvVars[key] = value
+		}
+	}
+
+	for key, value := range defaultEnvVars {
+		err := os.Setenv(key, value)
```

Comment:
I believe `testing.M` has a method for setting env vars in the context of the test, should we use that instead?

Reply:
Unfortunately , `testing.M` does not have a built in mechanism to set env variables, unlike `testing.T`. But since We need those Env variables in the `TestMain` function, there is no running away from using `os.setEnv`.